### ISSUE TITLE
Strengthen `unused-awaitable` for AnyIO and other frameworks where you always `await`

### ISSUE THREAD

**Issue by rdaysky** [astral-sh/ty#3820]
In AnyIO, curio, trio any invocation of an async function without an immediate await is an error, there are no `asyncio.gather(async_fn())`-like workflows.

There should be an opt-in warning for projects using frameworks like that enforcing the “always await” rule. It should differ from `unused-awaitable` in two ways:
- Not only a raw call discarding the return value, but any call of an async function is prohibited without an immediate `await`.
- It should not apply to expressions manipulating values that are already of CoroutineType. These are assumed to be low-level helpers where the developer knows what they’re doing.

Examples:

```py
async def f(): ...

def black_hole(x: Any): ...

def low_level_stuff[Y, S, R](c: Coroutine[Y, S, R]):
    black_hole(c)  # no warning

async def amain():
    f()  # warning
    black_hole(f())  # warning
    low_level_stuff(f())  # warning

    async with anyio.create_task_group() as tg:
        task_f = tg.start_soon(f)  # no warning

    p = functools.partial(f)  # or lambda f: f()
    black_hole(p())  # a warning would be great if ty can see through this
```
